2298 Fixes OCR extraction for recap documents

albertisfu opened the pull request:
This fixes OCR extraction for RECAP documents.

There were two problems in `extract_recap_pdf` that originated the issue:

- The recap document file was `read()` two times when needing OCR, the first time for `document-extract` and the second for `pdf-to-text`. When calling `read()` the end of the file is reached so the second time it returns an [empty string](https://docs.python.org/3/tutorial/inputoutput.html#methods-of-file-objects) so the request validation on Doctor failed and returned a `BAD_REQUEST` status code.

  So to solve it's needed to reopen the file on every request.  A context manager was used to open the file and read it.
Seems that the problem only was happening on `RECAPDocument` but I also applied the fix for `Opinion` and `Audio` so there's no problem if we ever make two or more requests in a row for these objects. 

- The second issue was using the OCR `response.content` that is bytes and passing to `anonymize()` that accepts `str` so instead of using `response.content` now we use `response.text`
- I added a test to verify OCR extraction, it's a bit slow but I think It would help us to detect any issue related to OCR extraction in the future.

Also, there's a fix for a test that is failing after Doctor status codes change.

mlissner commented on cl/recap/tests.py:
```diff
@@ -913,6 +917,17 @@ def test_nothing_already_exists(self) -> None:
+        cf = ContentFile(self.file_content_ocr)
+        self.pq.filepath_local.save(self.filename_ocr, cf)
+        process_recap_pdf(self.pq.pk)
+        recap_document = RECAPDocument.objects.all()
```
I forget exactly how tests and transactions work, but this feels fragile, right? Maybe make process_recap_pdf return the `rd`, or look up the item more carefully?

albertisfu replied:
Yeah, I've updated the test to look for the recap document more carefully.

I used a similar approach for `RecapEmailDocketAlerts` tests but seems it's safe since we're using [TestCase](https://docs.djangoproject.com/en/4.1/topics/testing/tools/#django.test.TestCase) so each test runs inside of a transaction, so they're isolated.

mlissner replied:
OK, I couldn't remember how those transactions work. Thanks for reminding me.

mlissner reviewed (changes requested):
A couple small comments. In addition, could you please add a small upgrade here?

I noticed over in doctor that the `/extract/pdf/text/` URL is really not needed. All of its functionality is also in the `/extract/doc/text/` URL. I'm working on a PR to remove the `/extract/pdf/text/` URL from doctor and do a release of that, but before we do that, we need to stop relying on it here. 

Can you refactor this one microservice call to use the other URL, and remove the /extract/pdf/text URL from our microservice function? 

Also, for the test being slow, can you shrink the image in the PDF so it's really small? That should speed it up significantly. It could even just contain a single word, I think, right?

mlissner commented:
FYI, I just pushed a couple little tweaks here to clean up unused code. Shouldn't affect anything.

albertisfu commented:
I've applied the changes and removed `/extract/pdf/text/` service and replaced with `/extract/doc/text/`.

Yeah, now I'm using a smaller PDF file for tests just with a small image, so the test runs faster.

Thank you!

mlissner commented:
Looks great, thanks. We could have made the PDF smaller too. I think that'd make it even faster. Like, making 100px by 50px. OCR looks at every pixel, so any you remove speeds things up. Not worth playing with more though.
